python/cudf/cudf/utils/utils.py

```diff
@@ -394,3 +394,10 @@ def pa_mask_buffer_to_mask(mask_buf, size):
         dbuf.copy_from_host(np.asarray(mask_buf).view("u1"))
         return Buffer(dbuf)
     return Buffer(mask_buf)
+
+
+def isnat(val):
+    if not isinstance(val, (np.datetime64, np.timedelta64)):
+        return False
+    else:
+        return val in {"NaT", "NAT"} or np.isnat(val)
```

Comment:
If we pass in `NaT` as a string, it will never hit the `else` branch here. Is that intended?

Reply:
Updated this.